Source/monster.cpp

```diff
@@ -4831,18 +4823,18 @@ bool LineClear(bool (*Clear)(int, Point), int entity, int x1, int y1, int x2, in
 			dincH = 2 * (dy + dx);
 			xincD = -1;
 		}
-		while (!done && (y1 != y2 || x1 != x2)) {
+		while (!done && (startPoint.y != endPoint.y || startPoint.x != endPoint.x)) {
 			if ((d <= 0) ^ (xincD < 0)) {
 				d += dincD;
 			} else {
 				d += dincH;
-				x1 += xincD;
+				startPoint.x += xincD;
 			}
-			y1++;
-			done = ((y1 != yorg || x1 != xorg) && !Clear(entity, { x1, y1 }));
+			startPoint.y++;
+			done = ((startPoint.y != yorg || startPoint.x != xorg) && !Clear(entity, startPoint));

```

Comment:
If xorg/yorg was also redefined this could be reduced to:
```suggestion
			done = (startPoint != origin && !Clear(entity, startPoint));
```